frontend/src/app/ml-models/ml-model-form/ml-model-form.component.ts

```diff
@@ -297,6 +299,67 @@ export class MlModelFormComponent implements OnInit {
     return true;
   }
 
+  /**
+   * Check the model variables selected to ensure the necessary variables have been
+   * specified and if not return any errors that would require user correction.
+   *
+   * @returns List of validation errors if any and otherwise returns undefined.
+   */
+  get variableErrors(): string[]|undefined {
```

Comment:
I'm on the fence if this should be required, but I think this method should be split up into a series of checks that this method calls.  Something like:

```
    let errors = []

    checkMinimumNumOfFeatures(errors);
    checkOnlyOneLabeWasSelected(errors);
    ...

    return errors.length > 0 ? errors : undefined;
```

This way, each function name describes the checks you're doing, and so it's easier for a reader to see what validation rules are being applied, without having to inspect each if / else statement and error message.